src/core/dyad_core.c

```diff
@@ -0,0 +1,622 @@
+#include "dyad_core.h"
+#include "dyad_err.h"
+#include "utils.h"
+#include "murmur3.h"
+
+#include <libgen.h>
+#include <limits.h>
+#include <string.h>
+#include <unistd.h>
+
+const struct dyad_ctx dyad_ctx_default = {
+    NULL,
+    false,
+    false,
+    false,
+    true,
+    false,
+    false,
+    3u,
+    1024u,
+    0u,
+    NULL,
+    NULL
+};
+
+int validate_context(dyad_ctx_t *ctx, bool check_cons, bool check_prod)
+{
+    if (!ctx || !ctx->h)
+    {
+        return DYAD_NOCTX;
+    }
+    if (check_cons && (ctx->cons_managed_path == NULL || strlen(ctx->cons_managed_path) == 0))
```

Comment:
Do it only once at the initialization if possible. If not, it is worth storing this result of this check and reuse it. `strlen()` will read every character in the string.